hls-plugin-api/src/Ide/Types.hs

```diff
@@ -693,6 +693,24 @@ nullToMaybe' :: (a |? (b |? Null)) -> Maybe (a |? b)
 nullToMaybe' (InL x)       = Just $ InL x
 nullToMaybe' (InR (InL x)) = Just $ InR x
 nullToMaybe' (InR (InR _)) = Nothing
+
+type Definitions = (Definition |? ([DefinitionLink] |? Null))
+
+mergeDefinitions :: Definitions -> Definitions -> Definitions
+mergeDefinitions definitions1 definitions2 = case (definitions1, definitions2) of
```

Comment:
There's are client capabilities controlling whether location links are supported. Ideally we should check those and if they're not sent we should instead downgrade the `LocationLink`s.

Reply:
I created a new `downgradeLinks` function instead and used it to downgrade all `LocationLink`s in the implementation of `combineResponses` when required by the missing capability.